pkg/task/task.go

```diff
@@ -0,0 +1,74 @@
+package task
+
+import "github.com/google/uuid"
+import "time"
+
+// TaskState (kind: int) represents the last known state of a task.
+// A task can be in one of three states
+// StateScheduled: this is the initial state of the task when it enters the queue.
+// StateProcessing: once work begins on the task, it is put into this state.
+// StateComplete: work is no longer being done on this task. client should check task result.
+type TaskState int
+
+const (
+	StateScheduled TaskState = iota
+	StateProcessing
+	StateComplete
+)
+
+// TaskType(kind int) represents the kind of activity the daemon asked to perform. In alignment
+// with the testground command-line we have two kinds of tasks
+// TaskBuild -- which functions similarly to `testground build`. The result of this task will contain
+// a build ID which can be used in a subsiquent run.
+// TaskRun -- which functions similarly to `testground run`
+type TaskType int
+
+const (
+	TaskBuild TaskType = iota
+	TaskRun
+)
+
+// DatedTaskState (kind: struct) is a TaskState with a timestamp.
+type DatedTaskState struct {
+	Created   time.Time `json:"created"`
+	TaskState TaskState `json:"state"`
+}
+
+type BuildResult struct {
+	ID string `json:"id"`
+}
+
+type RunResult struct {
+	Succeeded int `json:"succeeded"`
+	Running   int `json:"running"`
+	Pending   int `json:"pending"`
+	Failed    int `json:"failed"`
+	Unknown   int `json:"unknown"`
+}
+
+// TaskResult (kind: struct)  contains a status code. If the status is not TaskResultNone or
+// TaskResultSuccess, relevant errors will be included in this struct.
+type TaskResult struct {
+	BuildResult BuildResult `json:"build"`
+	RunResult   RunResult   `json:"run"`
+	Errors      []error     `json:"errors"`
+}
```

Comment:
As the comment below - I don't think the queue should know about these structs - they should be generic and dependent on the builder/runner, as far as the queue and the supervisor are concerned, we just have some `result`.

Reply:
For that matter, the task.States log and the Result could all be stored separately since they are not relevant to the queue